ui/components/app/signature-request-siwe/signature-request-siwe.js

```diff
@@ -30,6 +44,10 @@ export default function SignatureRequestSIWE({
 }) {
   const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
   const subjectMetadata = useSelector(getSubjectMetadata);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const currentChainId = useSelector(getCurrentChainId);
+  const provider = useSelector(getProvider);
```

Comment:
[out-of-scope] interesting...  wonder how much of this logic could be moved into `NetworkAccountBalanceHeader` itself